### PR TITLE
Add Dexcom One+ to allowed prefix

### PR DESCRIPTION
**Pull request by marionbarker**
I am not able to test this but this matches code used successfully by the iAPS community.

### CODE REVIEW

**Comment by marionbarker** [comment]
I modified the start up message to be more generic: "This app" instead of "Loop".

**Comment by Franeczek** [comment]
.

File: G7SensorKitUI/Views/G7StartupView.swift [hunk 1]
```diff
@@ -26,7 +26,7 @@ struct G7StartupView: View {
                     .frame(height: 120)
                     .padding(.horizontal)
             }.frame(maxWidth: .infinity)
-            Text(LocalizedString("Loop can read G7 CGM data, but you must still use the Dexcom G7 App for pairing, calibration, and other sensor management.", comment: "Descriptive text on G7StartupView"))
```
Comment (ps2) [inline G7SensorKitUI/Views/G7StartupView.swift:29]: What is the reason for this text change?

Comment (marionbarker) [reply G7SensorKitUI/Views/G7StartupView.swift:29]: The idea is that if you use a generic term for the app, a different open source app can use the same submodule and the translations would not require an update from the appName. 

Specifically, the OpenAPS implementation on the iPhone platform intends to use the same versions as are found in the LoopKit repositories.

Comment (marionbarker) [reply G7SensorKitUI/Views/G7StartupView.swift:29]: @ps2 Let me know if you want me to revert the language change.

Comment (marionbarker) [reply G7SensorKitUI/Views/G7StartupView.swift:29]: I only touched the hard coded “Loop”. Do you want me to add appName instead?

Comment (marionbarker) [reply G7SensorKitUI/Views/G7StartupView.swift:29]: Modified to use appName in text string

**Comment by ps2** [comment]
Did you test this?  Is appname being set for G7?

**Comment by marionbarker** [comment]
I tested modifying appName, but I'm sure I did it the "wrong" way.

* Modify `MAIN_APP_DISPLAY_NAME` in LoopConfigOverride.xcconfig to `appNameRename` and that is what is displayed as the display name for the app
   * after a phone reboot, that also shows up as name for app in Settings and Health Data Sources
   * But "Loop" is still shown as the appName in the G7SensorKit string

Modified LoopKit/LoopKitUI/Extensions/Environment+AppName.swift (line 14) from:
* `    static let defaultValue = "Loop"`
 to:
* `    static let defaultValue = "HardCodedSwift"`
* Now "HardCodedSwift" shows up in the G7SensorKit string

This is the screen with that change to the code and French selected as the language.

hard-coded-defaultValue-g7-french

**Comment by ps2** [comment]
App name needs to be injected into the environment, like this: `.environment(\.appName, Bundle.main.bundleDisplayName)`; search, and you'll see an example of that in the onboarding. Eventually we hope that this will just be done once at the Loop level, and not at each plugin, but since plugins vend UIKit views instead of SwiftUI views; we have to do this for now.

**Comment by marionbarker** [comment]
I modified the PR to just have the one on-topic modification.

**Pull request merged**